daemon/loadbalancer.go

```diff
@@ -182,47 +194,89 @@ func (h *deleteServiceID) Handle(params DeleteServiceIDParams) middleware.Respon
 	defer d.loadBalancer.BPFMapMU.Unlock()
 
 	svc, ok := d.loadBalancer.SVCMapID[types.ServiceID(params.ID)]
+
 	if !ok {
 		return NewDeleteServiceIDNotFound()
 	}
 
 	// FIXME: How to handle error?
-	DeleteL3n4AddrIDByUUID(uint32(params.ID))
+	err := DeleteL3n4AddrIDByUUID(uint32(params.ID))
+
+	if err != nil {
+		log.Warningf("error, DeleteL3n4AddrIDByUUID failed: %s", err)
+	}
 
 	if err := h.d.svcDelete(svc); err != nil {
+		log.Warningf("DELETE /service/{id}: error deleting svc %v: %s", svc, err)
 		return apierror.Error(DeleteServiceIDFailureCode, err)
 	}
 
 	return NewDeleteServiceIDOK()
 }
 
-// Deletes a service by the frontend address
-func (d *Daemon) svcDeleteByFrontend(frontend *types.L3n4Addr) error {
-	d.loadBalancer.BPFMapMU.Lock()
-	defer d.loadBalancer.BPFMapMU.Unlock()
-
+func (d *Daemon) svcDeleteByFrontendLocked(frontend *types.L3n4Addr) error {
 	svc, ok := d.loadBalancer.SVCMap[frontend.SHA256Sum()]
 	if !ok {
 		return fmt.Errorf("Service frontend not found %+v", frontend)
 	}
 	return d.svcDelete(&svc)
 }
 
+// Deletes a service by the frontend address
+func (d *Daemon) svcDeleteByFrontend(frontend *types.L3n4Addr) error {
+	d.loadBalancer.BPFMapMU.Lock()
+	defer d.loadBalancer.BPFMapMU.Unlock()
```

Comment:
This is confusing, can't we take this lock in the loadbalancer itself? Is this lock protecting the data structures in the daemon as well?

Reply:
I made this function to give flexibility when we need to delete a service by its `L3n4Addr` depending on if we have already acquired `BPFMapMU`. This is similar to other functions throughout the daemon. Can we sync up over video chat about this? It would be easier to explain that way.